src/components/structures/MatrixChat.js

```diff
@@ -1228,6 +1228,8 @@ module.exports = React.createClass({
     onReturnToGuestClick: function() {
         // reanimate our guest login
         if (this.state.guestCreds) {
+            // TODO: this is probably a bit broken - we don't want to be
+            // clearing storage when we reanimate the guest creds.
```

Comment:
Yep, in an ideal world we allow a user to go to `/login` without actually changing the `loggedIn` state of `MatrixChat`. At that point `onReturnToGuestClick` can just show `LoggedInView` without calling `Lifecycle`.